lib/modules/solidity/index.js

```diff
@@ -51,7 +53,12 @@ class Solidity {
         if (self.solcAlreadyLoaded) {
           return callback();
         }
-        self.solcW = new SolcW({logger: self.logger, events: self.events, ipc: self.ipc, useDashboard: self.useDashboard});
+        if (self.storageConfig) {
+          if(self.storageConfig.upload) {
```

Comment:
That's good. FYI, you can simplify it using `if (self.storageConfig && self.storageConfig.upload) {`
Also, can you put a similar condition in the other place where you get that config. Thanks.

Reply:
It works, because as soon as `self.storageConfig` is falsy, it stops reading the `if` (because it's  a `&&`)

Reply:
ok, understood.thanks